### ISSUE TITLE
Use pkg-config to detect glfw3 on Linux et al

### ISSUE THREAD

**Issue by wasamasa** [AlexCharlton/glfw3-chicken#4]
Spotted on `#chicken`:

```
<Menche> [03:43:13] can I modify how chicken-install links an egg?
<Menche> [03:44:09] the glfw3 egg looks for libglfw.so.3 but that's not installed here
<Menche> [03:44:23] only the static library libglfw3.a
```

It seems wrong to depend on a `libglfw.so.3` existing.  Please use `pkg-config glfw3 --libs` instead.

**Comment by AlexCharlton** [comment]
Thanks for spotting this! It's not personally a high priority, as this is the usual manner of linking libraries that don't support `pkg-config`, but I can see the utility in certain cases. If you feel like submitting a patch, it'd be appreciated!

**Comment by wasamasa** [comment]
```patch
--- a/glfw3.setup       2016-07-08 07:43:30.033360939 +0200
+++ b/glfw3.setup       2016-07-08 07:43:17.789697296 +0200
@@ -8,7 +8,7 @@
 
 (define link-options
   (cond-expand
-    ((and unix (not macosx)) "-l:libglfw.so.3")
+    ((and unix (not macosx)) (car (with-input-from-pipe "pkg-config glfw3 --libs" read-lines)))
     (macosx "-lglfw -framework Cocoa -framework OpenGL -framework IOKit -framework Corevideo")
     (windows "-lglfw -lopengl32 -lgdi32")
     (else "-lglfw")))
```

**Comment by AlexCharlton** [comment]
Super! Thanks @wasamasa!

Fixed in c6ae2587b30f450283ea7f942fa35002cde69782

New release tagged for this fix
